FirebaseDynamicLinks/Sources/FIRDynamicLinkNetworking.m

```diff
@@ -118,33 +119,65 @@ - (void)resolveShortLink:(NSURL *)url
     @"sdk_version" : FDLSDKVersion
   };
 
-  FIRNetworkRequestCompletionHandler resolveLinkCallback = ^(NSData *data, NSError *error) {
-    NSURL *resolvedURL;
-
-    if (!error && data) {
-      NSDictionary *result = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
-      if ([result isKindOfClass:[NSDictionary class]]) {
-        id invitationIDObject = [result objectForKey:@"invitationId"];
-
-        NSString *invitationIDString;
-        if ([invitationIDObject isKindOfClass:[NSDictionary class]]) {
-          NSDictionary *invitationIDDictionary = invitationIDObject;
-          invitationIDString = invitationIDDictionary[@"id"];
-        } else if ([invitationIDObject isKindOfClass:[NSString class]]) {
-          invitationIDString = invitationIDObject;
+  FIRNetworkRequestCompletionHandler resolveLinkCallback = ^(NSData *data, NSURLResponse *response,
+                                                             NSError *error) {
+    NSURL *resolvedURL = nil;
+
+    if (![response isKindOfClass:[NSHTTPURLResponse class]]) {
+      NSError *responseTypeError = [NSError
+          errorWithDomain:kGenericErrorDomain
+                     code:0
+                 userInfo:@{@"message" : @"Response should be of type NSHTTPURLResponse."}];
+      handler(resolvedURL, error ? error : responseTypeError);
```

Comment:
"responseTypeError" is used only if error is nil; so can we defer the responseTypeError error creation?

Reply:
Sure. Fixed.